scripts/build-controller-image.sh

```diff
@@ -70,15 +71,24 @@ if [[ $QUIET = "false" ]]; then
     echo " git commit: $SERVICE_CONTROLLER_GIT_COMMIT"
 fi
 
+# if workspace has local 'runtime' module
+# then use Dockerfile which allows references to local module from service controller go.mod file
+if [[ -d "${ACK_DIR}/runtime" ]]; then
+  DOCKER_BUILD_CONTEXT="$ACK_DIR"
```

Comment:
I don't quite understand why you need a separate build context for `Dockerfile.local`. Don't we want to keep the build context pointed at the service controller's directory?

Reply:
```
aws-controllers-k8s
├── code-generator
├── community
├── elasticache-controller
└── runtime
```
To copy the `runtime` during docker build, I supplied the build context as path leading upto `aws-controllers-k8s`

Reply:
I think what @A-Hilaly and I were saying is that the way this is currently written, if the scripts/build-controller-image.sh script is run by a developer locally when they have the set of repositories git clone'd, then the Dockerfile.local will *always* be used. And Amine and I were suggesting to use a specific `LOCAL` variable to trigger using the Dockerfile.local instead of just checking for whether the `../runtime` directory existed.

That way, I can run ./scripts/build-controller-image.sh locally (even with a git clone'd runtime repository) and build a controller image that uses the runtime version from its go.mod file.

Reply:
Considering that

> It is upto the `elasticache-controller/go.mod` to use runtime module from local workspace or from its github version; `Dockerfile.local` just makes the local runtime directory available during build.

The use of local vs remote runtime is controlled by the consumer module elasticache-controller/go.mod file and not by Dockerfile.localmodules as the presence of `runtime` module locally during docker build will not have any impact if the go.mod file refers it from remote.

Please validate.

Reply:
I had discussion with Jay on it. Per that discussion, considering following approach:
* Add a flag (example: `LOCAL_MODULES`) to select either `Dockerfile.local` or `Dockerfile`
* Supply the flag from makefile targets and name the targets to be explicit about the 'local' build (example: `make local-kind-test`)
* Expect service controller modules to have a `go.local.mod` and `go.local.sum` files for local builds
* Inside `Dockerfile.local` use `go.local.mod` file and `go.local.sum` files during `go build`

This PR has been updated per the above approach.

Reply:
Maybe i'm missing something here but I'm not sure about the `go.local.mod` and `go.local.sum` solution, how would users maintain their "local module files", is it possible to run `go mod tidy` and expect the `go.local.sum` to change? IMO it should be the responsibility of the developers to manage and maintain their modules/dependency files.

Reply:
If we are going with this solution, shall we add `go.local.mod` and `go.local.sum` to the `.gitignore` file?

Reply:
We should probably add `go.local.sum` to the .gitignore file, yeah. But `go.local.mod` should stay checked into git. This is the Go mod file that will have the `replace` directive(s).

Reply:
Added `elasticache-controller/go.local.sum` to `.gitignore` file, reference PR:
https://github.com/aws-controllers-k8s/elasticache-controller/pull/4